libs/vertexai/langchain_google_vertexai/_utils.py

```diff
@@ -138,6 +138,8 @@ def _missing_(cls, value: Any) -> "GoogleModelFamily":
             return GoogleModelFamily.PALM
         if "medlm-large" in value.lower():
             return GoogleModelFamily.PALM
+        elif "medlm" in value.lower():
```

Comment:
It's done on purpose, `medlm-medium@latest` is actually gemini family afaik (at least, API wise).
Could you adjust your PR accordingly, please?

Reply:
I just checked with the Google Gemini product team. They confirmed that the latest versions of the MedLM model are available in both medlm-medium and medlm-large sizes. They also directed me to this [documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/medlm/medlm-prompts) for API details and other information.

Without this fix, medlm-medium is classified as a Gemini family model and uses GenerativeModel, leading to predict call failures. The Google product team pointed out that I should use TextGenerationModel instead of GenerativeModel for both MedLM large & medium.

grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
status = StatusCode.RESOURCE_EXHAUSTED
details = "Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: MedLM-medium. Please submit a quota increase request. https://cloud.google.com/vertex-ai/docs/generative-ai/quotas-genai."
debug_error_string = "UNKNOWN:Error received from peer ipv4:10.6.65.85:8080 {created_time:"2024-06-06T19:51:53.145217318+00:00", grpc_status:8, grpc_message:"Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: MedLM-medium. Please submit a quota increase request. https://cloud.google.com/vertex-ai/docs/generative-ai/quotas-genai."}"
>
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/local/lib/python3.10/site-packages/chainlit/utils.py", line 40, in wrapper
return await user_function(**params_values)
File "/app/app.py", line 387, in main
res = await cl.make_async(agent.run)(input, callbacks=[cb])
File "/usr/local/lib/python3.10/site-packages/asyncer/_main.py", line 358, in wrapper
return await anyio.to_thread.run_sync(
File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 33, in run_sync
return await get_asynclib().run_sync_in_worker_thread(
File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
return await future
File "/usr/local/lib/python3.10/asyncio/futures.py", line 285, in __await__
yield self # This tells Task to wait for completion.
File "/usr/local/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup
future.result()
File "/usr/local/lib/python3.10/asyncio/futures.py", line 201, in result
raise self._exception.with_traceback(self._exception_tb)
File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
result = context.run(func, *args)
File "/usr/local/lib/python3.10/site-packages/langchain_core/_api/deprecation.py", line 148, in warning_emitting_wrapper
return wrapped(*args, **kwargs)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 600, in run
return self(args[0], callbacks=callbacks, tags=tags, metadata=metadata)[
File "/usr/local/lib/python3.10/site-packages/langchain_core/_api/deprecation.py", line 148, in warning_emitting_wrapper
return wrapped(*args, **kwargs)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 383, in __call__
return self.invoke(
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 166, in invoke
raise e
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 156, in invoke
self._call(inputs, run_manager=run_manager)
File "/usr/local/lib/python3.10/site-packages/langchain/agents/agent.py", line 1433, in _call
next_step_output = self._take_next_step(
File "/usr/local/lib/python3.10/site-packages/langchain/agents/agent.py", line 1139, in _take_next_step
[
File "/usr/local/lib/python3.10/site-packages/langchain/agents/agent.py", line 1139, in <listcomp>
[
File "/usr/local/lib/python3.10/site-packages/langchain/agents/agent.py", line 1167, in _iter_next_step
output = self.agent.plan(
File "/usr/local/lib/python3.10/site-packages/langchain/agents/agent.py", line 731, in plan
full_output = self.llm_chain.predict(callbacks=callbacks, **full_inputs)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/llm.py", line 316, in predict
return self(kwargs, callbacks=callbacks)[self.output_key]
File "/usr/local/lib/python3.10/site-packages/langchain_core/_api/deprecation.py", line 148, in warning_emitting_wrapper
return wrapped(*args, **kwargs)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 383, in __call__
return self.invoke(
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 166, in invoke
raise e
File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 156, in invoke
self._call(inputs, run_manager=run_manager)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/llm.py", line 126, in _call
response = self.generate([inputs], run_manager=run_manager)
File "/usr/local/lib/python3.10/site-packages/langchain/chains/llm.py", line 138, in generate
return self.llm.generate_prompt(
File "/usr/local/lib/python3.10/site-packages/langchain_core/language_models/llms.py", line 633, in generate_prompt
return self.generate(prompt_strings, stop=stop, callbacks=callbacks, **kwargs)
File "/usr/local/lib/python3.10/site-packages/langchain_core/language_models/llms.py", line 803, in generate
output = self._generate_helper(
File "/usr/local/lib/python3.10/site-packages/langchain_core/language_models/llms.py", line 670, in _generate_helper
raise e
File "/usr/local/lib/python3.10/site-packages/langchain_core/language_models/llms.py", line 657, in _generate_helper
self._generate(
File "/usr/local/lib/python3.10/site-packages/langchain_google_vertexai/llms.py", line 231, in _generate
res = _completion_with_retry(
File "/usr/local/lib/python3.10/site-packages/langchain_google_vertexai/llms.py", line 70, in _completion_with_retry
return _completion_with_retry_inner(prompt, is_gemini, **kwargs)
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 330, in wrapped_f
return self(f, *args, **kw)
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 467, in __call__
do = self.iter(retry_state=retry_state)
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 368, in iter
result = action(retry_state)
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 410, in exc_check
raise retry_exc.reraise()
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 183, in reraise
raise self.last_attempt.result()
File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 451, in result
return self.__get_result()
File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
raise self._exception
File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 470, in __call__
result = fn(*args, **kwargs)
File "/usr/local/lib/python3.10/site-packages/langchain_google_vertexai/llms.py", line 58, in _completion_with_retry_inner
return llm.client.generate_content(
File "/usr/local/lib/python3.10/site-packages/vertexai/generative_models/_generative_models.py", line 407, in generate_content
return self._generate_content(
File "/usr/local/lib/python3.10/site-packages/vertexai/generative_models/_generative_models.py", line 496, in _generate_content
gapic_response = self._prediction_client.generate_content(request=request)
File "/usr/local/lib/python3.10/site-packages/google/cloud/aiplatform_v1beta1/services/prediction_service/client.py", line 2125, in generate_content
response = rpc(
File "/usr/local/lib/python3.10/site-packages/google/api_core/gapic_v1/method.py", line 131, in __call__
return wrapped_func(*args, **kwargs)
File "/usr/local/lib/python3.10/site-packages/google/api_core/grpc_helpers.py", line 78, in error_remapped_callable
raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.ResourceExhausted: 429 Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: MedLM-medium. Please submit a quota increase request. https://cloud.google.com/vertex-ai/docs/generative-ai/quotas-genai.